src/accelerate/utils/modeling.py

```diff
@@ -1919,3 +1899,44 @@ def has_offloaded_params(module: torch.nn.Module) -> bool:
     from ..hooks import AlignDevicesHook  # avoid circular import
 
     return hasattr(module, "_hf_hook") and isinstance(module._hf_hook, AlignDevicesHook) and module._hf_hook.offload
+
+
+@contextlib.contextmanager
+def align_module(module: torch.nn.Module, execution_device: Optional[torch.device] = None):
```

Comment:
I wonder if the name could be more descriptive, like `align_module_device`.

Reply:
Maybe `align_module_params`?

Reply:
I'd be fine with that. I wanted to add `device` to the name as "align" does not imply this, at least not for me. It could also, for instance, be about aligning the dtype.

Reply:
There's a little bit of a lexical gap, as I haven't seen many uses of the word "onload". `align` comes from `AlignDevicesHook`, so on second thought I think I'd also prefer `align_module_device`